### PR TITLE
Aabb refactor

### PR DESCRIPTION
**Pull request by kvark**
r? @csherratt

### CODE REVIEW

File: src/aabb.rs [hunk 1]
```diff
@@ -327,39 +350,21 @@ impl<S: BaseFloat> Continuous<Point3<S>> for (Ray3<S>, Aabb3<S>) {
...
 
 impl<S: BaseFloat> Discrete for (Aabb2<S>, Aabb2<S>) {
-    // TODO: i don't like current implementation
     fn intersects(&self) -> bool {
-        let (ref a, ref b) = *self;
+        let (a0, a1) = (self.0.min(), self.0.max());
+        let (b0, b1) = (self.1.min(), self.1.max());
 
-        if (a.max().x <= b.min().x) || (a.min().x >= b.max().x) || (a.max().y <= b.min().y) ||
-            (a.min().y >= b.min().y)
```
Comment (kvark) [inline src/aabb.rs]: notice the actual bug in the old code: should be `b.max().y` on the right

File: src/lib.rs [hunk 2]
```diff
@@ -36,9 +37,9 @@ extern crate approx;
 extern crate cgmath;
 extern crate num;
 
-#[cfg(feature = "serde")]
+#[cfg(feature = "eders")]
```
Comment (csherratt) [inline src/lib.rs]: I'm not sure what you are doing here.

**Comment by kvark** [comment]
The usage of serde in the actual code is still gated by 'eders' feature, so it makes no sense to have the unused externals for when 'serde' is forcefully enabled but 'eders' is not.

**Comment by csherratt** [comment]
LGTM

**Pull request merged**